portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,43 +28,55 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import org.json.simple.JSONArray;
-
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   JSONArray messageList = new JSONArray();
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Sets the JSON array to the three hard messages
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
     messageList.clear();
-    messageList.add("Hello Naa'Oli!"); 
-    messageList.add("How are you doing Naa'Oli?");
-    messageList.add("Good to see you again Naa'Oli.");
 
-    // Sends the JSON array to the /data servlet
-    response.setContentType("text/html;");
+    for(Entity entity : results.asIterable()) {
+      messageList.add((String)entity.getProperty("Comment"));
+    }
+
+    response.setContentType("application/json;");
     response.getWriter().println(messageList.toString());
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get Input from the Form
     String preproccessedText = request.getParameter("user-input");
-    if(preproccessedText == null) {
+
+    if(request == null) {
```

Comment:
missing space after `if`. This should be: `if (request == null)`

https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
"Separating any reserved word, such as if, for or catch, from an open parenthesis (() that follows it on that line"

please fix this for other instances in this file (`for` and `if` primarily)